Microsoft.Toolkit.Uwp.SampleApp/SamplePages/CameraPreview/CameraPreviewPage.xaml.cs

```diff
@@ -0,0 +1,147 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Uwp.Helpers.CameraHelper;
+using Microsoft.Toolkit.Uwp.UI.Controls;
+using Microsoft.Toolkit.Uwp.UI.Extensions;
+using Windows.ApplicationModel;
+using Windows.Graphics.Imaging;
+using Windows.Media;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media.Imaging;
+using Windows.UI.Xaml.Navigation;
+
+namespace Microsoft.Toolkit.Uwp.SampleApp.SamplePages
+{
+    /// <summary>
+    /// CameraPreviewPage
+    /// </summary>
+    public sealed partial class CameraPreviewPage : Page, IXamlRenderListener
+    {
+        private static SemaphoreSlim semaphoreSlim;
+        private VideoFrame _currentVideoFrame;
+        private SoftwareBitmapSource _softwareBitmapSource;
+        private CameraPreview _cameraPreviewControl;
+        private Image _imageControl;
+        private TextBlock _errorMessageText;
+
+        public CameraPreviewPage()
+        {
+            this.InitializeComponent();
+            semaphoreSlim = new SemaphoreSlim(1);
+        }
+
+        public async void OnXamlRendered(FrameworkElement control)
+        {
+            // Using a semaphore lock for synchronocity. 
+            // This method gets called multiple times when accessing the page from Latest Pages
+            // and creates unused duplicate references to Camera and memory leaks.
+            await semaphoreSlim.WaitAsync();
```

Comment:
Instead of using semaphore, why not create one CameraHelper and attach it to the CameraPreview every time it's rendered?